cartography/intel/aws/ec2/network_interfaces.py

```diff
@@ -22,191 +22,229 @@ def get_network_interface_data(boto3_session, region):
-    netinf.id = network_interface.NetworkInterfaceId, netinf.private_dns_name = network_interface.PrivateDnsName,
-    netinf.status = network_interface.Status, netinf.subnetid = network_interface.SubnetId,
-    netinf.interface_type = network_interface.InterfaceType,
-    netinf.requester_managed = network_interface.RequesterManaged,
-    netinf.source_dest_check = network_interface.SourceDestCheck,
-    netinf.requester_id = network_interface.RequesterId, netinf.public_ip = network_interface.Association.PublicIp
+        MERGE (netinf:NetworkInterface{id: network_interface.NetworkInterfaceId})
+        ON CREATE SET netinf.firstseen = timestamp()
+        SET netinf.lastupdated = {aws_update_tag},
+            netinf.mac_address = network_interface.MacAddress,
+            netinf.description = network_interface.Description,
+            netinf.private_ip_address = network_interface.PrivateIpAddress,
+            netinf.id = network_interface.NetworkInterfaceId,
+            netinf.private_dns_name = network_interface.PrivateDnsName,
+            netinf.status = network_interface.Status,
+            netinf.subnetid = network_interface.SubnetId,
+            netinf.interface_type = network_interface.InterfaceType,
+            netinf.requester_managed = network_interface.RequesterManaged,
+            netinf.source_dest_check = network_interface.SourceDestCheck,
+            netinf.requester_id = network_interface.RequesterId,
+            netinf.public_ip = network_interface.Association.PublicIp
+        WITH network_interface, netinf
+
+        UNWIND network_interface.PrivateIpAddresses AS private_ip_address
+            MERGE (private_ip:EC2PrivateIp{id: network_interface.NetworkInterfaceId + ':'
+                + private_ip_address.PrivateIpAddress})
+            ON CREATE SET private_ip.firstseen = timestamp()
+            SET private_ip.lastupdated = {aws_update_tag},
+                private_ip.network_interface_id = network_interface.NetworkInterfaceId,
+                private_ip.primary = private_ip_address.Primary,
+                private_ip.private_ip_address = private_ip_address.PrivateIpAddress,
+                private_ip.public_ip = private_ip_address.Association.PublicIp,
+                private_ip.ip_owner_id = private_ip_address.Association.IpOwnerId
+
+            MERGE (netinf)-[r:PRIVATE_IP_ADDRESS]->(private_ip)
+            ON CREATE SET r.firstseen = timestamp()
+            SET r.lastupdated = {aws_update_tag}
+        WITH network_interface, netinf
+
+        UNWIND network_interface.Groups AS security_group
+            MERGE (sg:EC2SecurityGroup{id: security_group.GroupId})
+            MERGE (netinf)-[r:MEMBER_OF_EC2_SECURITY_GROUP]->(sg)
+            ON CREATE SET r.firstseen = timestamp()
+            SET r.lastupdated = {aws_update_tag}
+        WITH network_interface, netinf
+
+        MERGE (acc:AWSAccount{id: {aws_account_id}})
+        MERGE (acc)-[r:RESOURCE]->(netinf)
```

Comment:
I am tying network interfaces to the AWS accounts that they belong to so that it is easier to clean them up: simply query for all netinfs attached to the current account and delete the stale ones.